lib/endpoints/metrics.js

```diff
@@ -17,12 +17,50 @@ var mod_restify = require('restify');
 
 var lib_errors = require('../errors');
 
+function fetchAndSendMetrics(req, res, req2, next) {
+    req2.on('result', function _fetchMetrics(err2, res2) {
+        if (err2) {
+            req.log.error(err2, 'Error fetching metrics');
+            next(new lib_errors.InternalServerError());
+            return;
+        }
+
+        var encoding;
+        if ((encoding = res2.header('Content-Encoding'))) {
+            res.header('Content-Encoding', encoding);
+        }
+
+        res.header('content-type', 'text/plain');
+
+        var data = [];
+        res2.on('data', function _onData(chunk) {
+            data.push(chunk);
+        }).on('end', function _onEnd() {
+            var buffer = Buffer.concat(data);
+            res.end(buffer);
+            next();
+        });
+    });
+}
+
 function apiGetMetrics(req, res, next) {
     var serverCacheItem, admin_ip, cmon_agent_url, cmon_agent_path, client;
 
     var target_uuid = req.header('HOST').split('.')[0];
     var cache = req.app.cache;
+    var config = req.app.config;
+
     mod_assert.object(cache, 'cache object');
+    mod_assert.object(config, 'config object');
+
+    var encoding;
+    var headers = {};
+    var server_opts = config.server_opts;
+    if (server_opts.http_accept_encoding === 'enabled') {
+         if ((encoding = req.header('Accept-Encoding'))) {
+             headers['Accept-Encoding'] = encoding;
```

Comment:
It would be nice to add a comment here that describes the restify behavior when `Accept-Encoding` is non-null. Also, it's not clear to me from the documentation, but do we also need to enable the gzip plugin on the restify server?

Reply:
I will add a comment describing what is going on here.

I don't think we need to enable gzip plugin in cmon because we are not doing compression on cmon side. We are simply passing what we get from cmon-agent as it is, regardless of whether it is compressed or not, and we are copying `Content-Encoding` from cmon-agent response to tell cmon clients whether what we got was compressed or not.

Reply:
👍